lib/client/api.go

```diff
@@ -2715,37 +2733,55 @@ func (tc *TeleportClient) ListKubernetesClustersWithFiltersAllClusters(ctx conte
 func (tc *TeleportClient) runCommandOnNodes(
 	ctx context.Context, siteName string, nodeAddresses []string, proxyClient *ProxyClient, command []string,
 ) error {
-	resultsC := make(chan error, len(nodeAddresses))
+	clt, err := proxyClient.ConnectToCluster(ctx, siteName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer clt.Close()
+
+	// Let's check is first node has mfa required.
+	// If it's required, run commands sequentially to avoid
+	// race conditions and weird ux during mfa.
+	mfaRequiredCheck, err := clt.IsMFARequired(ctx, &proto.IsMFARequiredRequest{
+		Target: &proto.IsMFARequiredRequest_Node{
+			Node: &proto.NodeLogin{
+				// Let's check only single node, all must have the same settings.
+				// TODO(rosstimothy): is it right?
```

Comment:
@rosstimothy can you confirm it?

Reply:
@rosstimothy maybe I am missing something but I don't think that  this check has been performed before. 
Connecting to node is happening after I am checking if this is required: https://github.com/gravitational/teleport/blob/408ac68f934d11014c93571270a695fd2e6f6dd1/lib/client/api.go#L2745-L2767
Moreover this check happens only if this ssh command should be executed on multiple ssh nodes, so I guess user can accept that additional latency.

Reply:
@rosstimothy I have added estabilishing connection to auth before as separate commit, but to be honest I don't like. ConnectToNode is used in a lot of places and we can either:
1. always dial auth and pass it to every `ConnectToNode`
2. or pass client as optional (i don't like passing client as optional).
3. Add new function `ConnectToNodeWithAuthClient` and do the same with ` proxy.IssueUserCertsWithMFA` and `sessionSSHCertificate`. This seems also like suboptimal solution.

So I have decided go with 1. But I am afraid that it may break something, because now we are doing that dial much earlier. UT are passing however.
One place where it can fail is https://github.com/gravitational/teleport/blob/0532344905800b6fdf3d497509478df5b70dbfff/lib/client/client.go#L2143-L2150. It was always executed before dialing to cluster auth. 🤔 

I ahve also tried passing only info about required check but it is also very ugly, a lot of ifs in `proxy.IssueUserCertsWithMFA` and we need to pass that parameter or optional in a lot places.

So to sum up, i would prefer that additional latency only on running ssh on multiple host. But if you still think we should go with passing auth client, please review my newest commit.

Reply:
> What are your thoughts about renaming NodeAddr to NodeDetails and adding a field to that struct and to ReissueParams which indicates if mfa is required?

Here we have 2 paths:
1. Always require that `is_mfa_required` which will result in changing it in a lot places. 
2. Allow passing it as optional parameter - but it means that is needs to be `required|notRequired|unknown` and it makes `proxy.IssueUserCertsWithMFA` not readable.

Solution 1 here is very similar that what I did with calling auth earlier (because that auth will be necessarny anyway to check `is_mfa_required`. So after all I think that solution which is implemented now (with passing auth client) is better. 
What do you think?

Reply:
@rosstimothy should be ready to re-review